File: Code/FrmBigPrimeGeneratorCs/Services.cs
//Copyright(c) update 2024 the exc-jdbi
//All rights reserved, see license.
//https://github.com/exc-jdbi/BigPrimeGenerators


using System.Text;

namespace exc.jdbi.VeryBigPrimes.Generators;


public class ServicesUtils
{

  //Es macht keinen Sinn OFD und SFD asynchron zu machen.
  //Das Speichern der Daten wie auch das Lesen hingegen schon.

  ////Beide Asynchrone Methoden funktionieren nicht, da der HauptThread
  ////verlassen wird.
  //public async static Task SaveDataAsync(FrmMain frm, string savetxt) =>
  //  await Task.Run(() => SaveData(frm, savetxt));

  //public async static Task<string> LoadDataAsync(OpenFileDialog ofd) =>
  //  await Task.Run(() => LoadData(ofd));

  public static void SaveDataSFD(string savetxt)
  {
    using var sfd = new SaveFileDialog();
    {
      sfd.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*";
      sfd.FilterIndex = 2;
      sfd.InitialDirectory = Application.StartupPath;
      sfd.RestoreDirectory = true;
      if (sfd.ShowDialog() == DialogResult.OK)
      {
        using var mystream = sfd.OpenFile();
        if (mystream != null)
        {
          var buffer = Encoding.UTF8.GetBytes(savetxt);
          mystream.Write(buffer, 0, buffer.Length);
          mystream.Close();
        }
      }
    }
  }

  public static string LoadDataOFD()
  {
    var result = string.Empty;
    {
      using var ofd = new OpenFileDialog();
      ofd.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*";
      ofd.FilterIndex = 2;
      ofd.InitialDirectory = Application.StartupPath;
      ofd.RestoreDirectory = true;
      try
      {
        if (ofd.ShowDialog() == DialogResult.OK)
        {
          using var mystream = ofd.OpenFile();
          if (mystream != null)
          {
            var len = Convert.ToInt32(mystream.Length);
            var buffer = new byte[len - 1 + 1];
            mystream.Read(buffer, 0, buffer.Length);
            result = Encoding.UTF8.GetString(buffer);
          }
        }
      }
      catch (Exception ex)
      {
        MessageBox.Show($"Can not read file: {ex}", "Info PFPG System");
      }
    }
    return result;
  }
}

File: Code/FrmBigPrimeGeneratorVb/Services.vb
'Copyright(c) update 2024 the exc-jdbi
'All rights reserved, see license.
'https://github.com/exc-jdbi/BigPrimeGenerators

Option Strict On
Option Explicit On

Imports System.Text
Imports System.Windows.Forms

Namespace exc.jdbi.VeryBigPrimes.Generators
  Friend Module ServicesUtils
    Public Sub SaveDataSFD(savetxt As String)
      Using sfd = New SaveFileDialog()

        If True Then
          sfd.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*"
          sfd.FilterIndex = 2
          sfd.InitialDirectory = Application.StartupPath
          sfd.RestoreDirectory = True
          If sfd.ShowDialog() = DialogResult.OK Then
            Using mystream = sfd.OpenFile()
              If mystream IsNot Nothing Then
                Dim buffer = Encoding.UTF8.GetBytes(savetxt)
                mystream.Write(buffer, 0, buffer.Length)
                mystream.Close()
              End If
            End Using
          End If
        End If
      End Using
    End Sub

    Public Function LoadDataOFD() As String
      Dim result = String.Empty
      If True Then
        Using ofd = New OpenFileDialog()
          ofd.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*"
          ofd.FilterIndex = 2
          ofd.InitialDirectory = Application.StartupPath
          ofd.RestoreDirectory = True
          Try
            If ofd.ShowDialog() = DialogResult.OK Then
              Using mystream = ofd.OpenFile()
                If mystream IsNot Nothing Then
                  Dim len = Convert.ToInt32(mystream.Length)
                  Dim buffer = New Byte(len - 1 + 1 - 1) {}
                  mystream.Read(buffer, 0, buffer.Length)
                  result = Encoding.UTF8.GetString(buffer)
                End If
              End Using
            End If
          Catch ex As Exception
            MessageBox.Show($"Can not read file: {ex}", "Info PFPG System")
          End Try
        End Using
      End If
      Return result
    End Function
  End Module
End Namespace
